src/cjson/lua_cjson.c

```diff
@@ -170,11 +170,19 @@ typedef struct {
     int decode_array_with_array_mt;
 } json_config_t;
 
+typedef struct {
+    /* convert null in json objects to lua nil instead of vim.NIL */
+    int object_use_luanil;
+    /* convert null in json arrays to lua nil instead of vim.NIL */
+    int array_use_luanil;
+} json_options_t;
```

Comment:
how is this different from `json_config_t` ?

If it's intended to separate our "fork", eh... at least name it `json_config2_t`. Don't choose a similar-but-different name if it's just extending something. But also I don't think we should try to separate our "fork" stuff because that implies we are going to be doing a lot more forking, YAGNI.

Reply:
`json_config_t` is set at the time of initializing the cjson module, these options do not change. `json_options_t` is set at the time of the function call.